projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,286 @@
+
+    /** (optional) Text that be shown when the Regular Expression condition not met */
+    @Prop() validationMessage = "The field doesn't comply with the specified format";
+
+    @Prop() URLValidationMessage = 'The specified URL is not valid';
+
+    /** (optional) Disables field's interaction */
+    @Prop({ reflectToAttr: true })
+    disabled = false;
+
+    /** (optional) Describes a type of file that may be selected by the user, separated by comma  eg: .pdf,.jpg  */
+    @Prop({ reflectToAttr: true })
+    accept = '';
+
+    /** (optional) Text that be shown in the browse file button */
+    @Prop({ reflectToAttr: true })
+    buttonLabel = 'Browse';
+
+    @State() status: DotFieldStatus;
+
+    @Event() valueChange: EventEmitter<DotFieldValueEvent>;
+    @Event() statusChange: EventEmitter<DotFieldStatusEvent>;
+
+    private allowedFileTypes = [];
+    private binaryTextValue = '';
+    private errorMessage = '';
+    private binaryTextField: DotBinaryTextField;
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     */
+    @Method()
+    reset(): void {
+        this.value = '';
+        this.binaryTextField.reset();
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    componentWillLoad(): void {
+        this.validateProps();
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+    }
+
+    componentDidLoad(): void {
+        this.binaryTextField = this.el.querySelector('dot-binary-text-field');
+    }
+
+    @Watch('accept')
+    optionsWatch(): void {
+        const validTypes = checkProp<DotBinaryFileComponent, string>(this, 'accept');
+        this.allowedFileTypes = !!validTypes ? validTypes.split(',') : [];
+    }
+
+    @Listen('_valueChange')
+    binaryTextValueChange(event: CustomEvent) {
+        event.stopImmediatePropagation();
+        this.setFileDisplayName(this.binaryTextField.value);
+        const eventValue: DotFieldValueEvent = event.detail;
+        this.setValue(eventValue.value);
+    }
+
+    @Listen('_statusChange')
+    binaryTextStatusChange(event: CustomEvent) {
+        event.stopImmediatePropagation();
+        const statusEvent: DotBinaryTextStatusEvent = event.detail;
+        this.errorMessage = statusEvent.errorType;
+        this.status = statusEvent.status;
+        if (this.status.dotPristine && this.status.dotTouched) {
+            this.emitStatusChange();
+        }
+    }
+
+    @Listen('dragover')
+    HandleDragover(): void {
+        if (!this.disabled) {
+            this.el.classList.add('dot-dragover');
+            this.el.classList.remove('dot-dropped');
+        }
+    }
+
+    @Listen('dragleave')
+    HandleDragleave(): void {
+        this.el.classList.remove('dot-dragover');
+        this.el.classList.remove('dot-dropped');
+    }
+
+    @Listen('drop', { passive: false })
```

Comment:
what is `passive` ?

Reply:
https://stenciljs.com/docs/events#passive